tests/Functional/Console/Command/RunCommandTest.php

```diff
@@ -111,11 +111,117 @@ public function testDumpXml()
         $this->assertContains('Dumped', $display);
         $this->assertTrue(file_exists(self::TEST_FNAME));
     }
+
+    /**
+     * It should dump to stdout.
+     */
+    public function testDumpXmlStdOut()
+    {
+        $tester = $this->runCommand('run', array(
+            '--dump' => true,
+            'path' => __DIR__ . '/../../benchmarks/BenchmarkCase.php',
+        ));
+        $this->assertEquals(0, $tester->getStatusCode());
+        $display = $tester->getDisplay();
+        $dom = new \DOMDocument();
+        $dom->loadXml($display);
+        $xpath = new \DOMXPath($dom);
+        $benchmarkEls = $xpath->query('//subject');
+        $this->assertEquals(3, $benchmarkEls->length);
+    }
+
+    /**
+     * It should accept explicit parameters.
+     */
+    public function testOverrideParameters()
+    {
+        $tester = $this->runCommand('run', array(
+            '--dump' => true,
+            '--parameters' => '{"length": 333}',
+            'path' => __DIR__ . '/../../benchmarks/BenchmarkCase.php',
+        ));
+        $this->assertEquals(0, $tester->getStatusCode());
+        $display = $tester->getDisplay();
+        $dom = new \DOMDocument();
+        $dom->loadXml($display);
+        $xpath = new \DOMXPath($dom);
+        $parameters = $xpath->query('//parameter[@value=333]');
+        $this->assertEquals(3, $parameters->length);
+    }
+
+    /**
+     * It should throw an exception if an invalid JSON string is provided for parameters.
+     *
+     * @expectedException InvalidArgumentException
+     */
+    public function testOverrideParametersInvalidJson()
+    {
+        $this->runCommand('run', array(
+            '--dump' => true,
+            '--parameters' => '{"length: 3,',
+            'path' => __DIR__ . '/../../benchmarks/BenchmarkCase.php',
+        ));
+    }
+
+    /**
+     * Its should allow the number of iterations to be specified.
+     */
+    public function testOverrideIterations()
+    {
+        $tester = $this->runCommand('run', array(
+            '--subject' => 'benchRandom',
+            '--dump' => true,
+            '--iterations' => 10,
+            'path' => __DIR__ . '/../../benchmarks/BenchmarkCase.php',
+        ));
+
+        $this->assertEquals(0, $tester->getStatusCode());
+        $display = $tester->getDisplay();
+        $dom = new \DOMDocument();
+        $dom->loadXml($display);
+        $dom->formatOutput = true;
+        $xpath = new \DOMXPath($dom);
+        $parameters = $xpath->query('//iteration');
+        $this->assertEquals(10, $parameters->length);
+    }
+
+    /**
+     * It can run each iteration in isolation.
+     */
+    public function testProcessIsolationIteration()
+    {
+        $this->runCommand('run', array(
+            '--processisolation' => 'iteration',
+            'path' => __DIR__ . '/../../benchmarks/BenchmarkCase.php',
+        ));
+    }
+
+    /**
+     * It can run each set of iterations in isolation.
+     */
```

Comment:
How to ensure separate processes were spawned? Write PID to `/tmp` from the test case?